tests/compile-fail/closure_called_early.rs

```diff
@@ -0,0 +1,7 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(closure_called_early)]
+fn main() {
+	let a = (|| 42)();
+}
```

Comment:
I'd like to see more tests, both with arguments to the closure, a closure as argument, and a closure defined in a `let` statement and directly used thereafter (we could choose to extend the lint to also cover otherwise unused closures from local vars, but for now a test can document that we don't)